pkg/occlient/occlient_test.go

```diff
@@ -99,6 +100,121 @@ func fakePodStatus(status corev1.PodPhase, podName string) *corev1.Pod {
+				dc: &appsv1.DeploymentConfig{
+					Spec: appsv1.DeploymentConfigSpec{
+						Selector: map[string]string{
+							"deploymentconfig": "nodejs-app",
+						},
+						Template: &corev1.PodTemplateSpec{
+							Spec: corev1.PodSpec{
+								Containers: []corev1.Container{
+									{
+										Name: "test",
+										VolumeMounts: []corev1.VolumeMount{
+											{
+												MountPath: "/tmp",
+												Name:      "test",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				pvc:  "testglocks volume",
+				path: "/mnt",
+			},
+			wantErr: false,
+		},
+		{
+			name: "Test case: dc without Containers defined",
+			args: args{
+				dc: &appsv1.DeploymentConfig{
+					Spec: appsv1.DeploymentConfigSpec{
+						Selector: map[string]string{
+							"deploymentconfig": "nodejs-app",
+						},
+						Template: &corev1.PodTemplateSpec{
+							Spec: corev1.PodSpec{},
+						},
+					},
+				},
+				pvc:  "my xx voulme",
+				path: "/mnt",
+			},
+			wantErr: true,
+		},
+		{
+			name: "Test case: dc without Template defined",
+			args: args{
+				dc: &appsv1.DeploymentConfig{
+					Spec: appsv1.DeploymentConfigSpec{
+						Selector: map[string]string{
+							"deploymentconfig": "nodejs-app",
+						},
+					},
+				},
+				pvc:  "my xx voulme",
+				path: "/mnt",
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakeClient, fakeClientset := FakeNew()
+
+			fakeClientset.AppsClientset.PrependReactor("update", "deploymentconfigs", func(action ktesting.Action) (bool, runtime.Object, error) {
+				return true, nil, nil
+			})
+			err := fakeClient.AddPVCToDeploymentConfig(tt.args.dc, tt.args.pvc, tt.args.path)
+
+			//Checks for error in positive cases
+			if !tt.wantErr == (err != nil) {
+				t.Errorf("Client.AddPVCToDeploymentConfig() unexpected error = %v, wantErr %v", err, tt.wantErr)
+			}
+
+			// Checks for number of actions performed in positive cases
+			if err == nil {
+				// Check for validating actions performed
+				if (len(fakeClientset.AppsClientset.Actions()) != 1) && (tt.wantErr != true) {
+					t.Errorf("expected 1 action in GetPVCFromName got: %v", fakeClientset.AppsClientset.Actions())
+				}
+				updatedDc := fakeClientset.AppsClientset.Actions()[0].(ktesting.UpdateAction).GetObject().(*appsv1.DeploymentConfig)
+				found := false
+				for bb := range updatedDc.Spec.Template.Spec.Containers[0].VolumeMounts {
+					if tt.args.path == updatedDc.Spec.Template.Spec.Containers[0].VolumeMounts[bb].MountPath {
+						found = true
+						if !strings.Contains(updatedDc.Spec.Template.Spec.Containers[0].VolumeMounts[bb].Name, tt.args.pvc) {
+							t.Errorf("pvc name not matching with the specived value got: %v, expected %v", updatedDc.Spec.Template.Spec.Containers[0].VolumeMounts[bb].Name, tt.args.pvc)
+						}
+					}
+				}
+				if found == false {
+					t.Errorf("expected Volume mount path %v not found", tt.args.path)
+				}
```

Comment:
We should also have a test for this part of the code i.e the volumes, check the volume name as before and that the claimName is the same as desired
https://github.com/redhat-developer/odo/blob/4cd7e795bec7979b74e9856d95c8b18a1c663629/pkg/occlient/occlient.go#L1339-L1346

Reply:
added validation for this